### PR TITLE
Test cases for Unbox Generic types

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

File: test/Streamly/Test/Data/Unbox.hs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+DERIVE_UNBOX(Product2)
+
+data SumOfProducts = SOP0 | SOP1 Int | SOP2 Int Char | SOP3 Int Int Int deriving (Show, Generic, Eq)
+DERIVE_UNBOX(SumOfProducts)
+
+data NestedSOP = NSOP0 SumOfProducts | NSOP1 SumOfProducts deriving (Show, Generic, Eq)
+DERIVE_UNBOX(NestedSOP)
+
+moduleName' :: String
+moduleName' = "Data.Unboxed"
```
Comment (harendra-kumar) [inline test/Streamly/Test/Data/Unbox.hs:66]: Why is there a prime at the end of the name?

**Review by harendra-kumar — APPROVED** [review]

File: test/Streamly/Test/Data/Unbox.hs (hunk 1 above)
Comment (rnjtranjan) [reply test/Streamly/Test/Data/Unbox.hs:66]: "moduleName" was conflicting with "Generic" module.

Comment (harendra-kumar) [reply test/Streamly/Test/Data/Unbox.hs:66]: You could import Generic hiding `moduleName`.

File: test/Streamly/Test/Data/Unbox.hs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    x2 <- run $  IORef.readIORef ref
+    assert (x2 == v2)
+    assert (x1 == v1)
+
+checkSizeOf :: (SizeOfRep (Rep a)) => Proxy a -> Int -> Property
+checkSizeOf v size = monadicIO $
+    assert (genericSizeOf v == size)
+
+data Unit = Unit deriving (Show, Generic, Eq)
+DERIVE_UNBOX(Unit)
```
Comment (harendra-kumar) [inline test/Streamly/Test/Data/Unbox.hs:51]: Now that we have default methods. We can use `instance Unbox Unit` here and remove the DERIVE_UNBOX macro.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Review by rnjtranjan — COMMENTED** [review]
Looks good.

**Pull request merged**